src/Utilities/useConnectionStatus.js

```diff
@@ -0,0 +1,29 @@
+import { useAxiosWithPlatformInterceptors } from '@redhat-cloud-services/frontend-components-utilities/interceptors';
+const { API_BASE } = require('../config');
+import { useState, useEffect } from 'react';
+
+export const useConnectionStatus = (remediation) => {
+  const axios = useAxiosWithPlatformInterceptors();
+  const [isConnected, setisConnected] = useState();
+
+  useEffect(() => {
```

Comment:
To be safe and not cause any unexpected errors or behaviours it might be good if we add a similar guard as here[0], preventing the state update when it's unmounted.

The gist of it is to have a Ref (`mounted` in the example) with a boolean that is set to false by default, set to true in the beginning of the `useEffect` and set to false in the cleanup function. 

[0] https://github.com/RedHatInsights/patchman-ui/pull/1153/files